sql-plugin/src/main/java/com/nvidia/spark/rapids/GpuColumnVector.java

```diff
@@ -895,6 +895,22 @@ public static ColumnarBatch combineColumns(ColumnarBatch ... batches) {
     return incRefCounts(ret);
   }
 
+  public static ColumnarBatch combineColumns(ColumnarBatch cb, GpuColumnVector ... vectors) {
+    final int numRows = cb.numRows();
+    ArrayList<ColumnVector> columns = new ArrayList<>();
```

Comment:
Nit: We know exactly how big the array needs to be and can hint this here.
```suggestion
    ArrayList<ColumnVector> columns = new ArrayList<>(cb.numCols() + vectors.length);
```